lib/package-manager/install.js

```diff
@@ -27,17 +28,26 @@ function install(packageManager, context, next) {
     `Using temp directory: "${options.env['npm_config_tmp']}"`
   );
 
-  if (context.module.install) {
-    args = args.concat(context.module.install);
+  let nodeBin = 'node';
+  if (context.options.testPath) {
+    options.env.PATH = `${context.options.testPath}:${process.env.PATH}`;
+    nodeBin = which('node', {
+      path: options.env.PATH || process.env.PATH
+    });
   }
 
   const packageManagerBin =
     packageManager === 'npm' ? context.npmPath : context.yarnPath;
 
+  let args = [packageManagerBin, 'install'];
+  if (context.module.install) {
+    args = args.concat(context.module.install);
+  }
+
   const binDirectory = path.dirname(packageManagerBin);
   options.env.PATH = `${binDirectory}:${process.env.PATH}`;
 
-  const proc = spawn(packageManagerBin, args, options);
+  const proc = spawn(nodeBin, args, options);
```

Comment:
does this still work if the package manager is a script? Say I halve aliased `npm` to some

```sh
#!/usr/bin/env bash

node /path/to/npm.js --log-level=silly $0
```

---

More real script - the recommended way of installing yarn on mac is through brew - and they use a script not a JS file. So node cannot execute the file. See https://github.com/nodejs/citgm/pull/560#issuecomment-432601911